clang/runtime/dpct-rt/include/dpct/group_utils.hpp

```diff
@@ -517,16 +516,115 @@ class workgroup_load {
   __dpct_inline__ void load(const Item &item, InputIteratorT block_itr,
                             InputT (&items)[ITEMS_PER_WORK_ITEM]) {
 
-    if constexpr (ALGORITHM == BLOCK_LOAD_DIRECT) {
+    if constexpr (ALGORITHM == load_algorithm::BLOCK_LOAD_DIRECT) {
       load_blocked<ITEMS_PER_WORK_ITEM>(item, block_itr, items);
-    } else if constexpr (ALGORITHM == BLOCK_LOAD_STRIPED) {
+    } else if constexpr (ALGORITHM == load_algorithm::BLOCK_LOAD_STRIPED) {
       load_striped<ITEMS_PER_WORK_ITEM>(item, block_itr, items);
     }
   }
 
 private:
   uint8_t *_local_memory;
 };
+
+/// Store blocked/warped or striped work items into linear segment of items.
+/// Helper for Block Store
+enum class store_algorithm {
+
+  BLOCK_STORE_DIRECT,
+  BLOCK_STORE_STRIPED,
+
+};
+
+/// Stores a blocked arrangement of work items linear segment of items.
+template <size_t ITEMS_PER_WORK_ITEM, typename InputT, typename OutputIteratorT,
+          typename Item>
+__dpct_inline__ void store_blocked(const Item &item, OutputIteratorT block_itr,
+                                   InputT (&items)[ITEMS_PER_WORK_ITEM]) {
+
+  // This implementation does not take in account range storage across
+  // workgroup items To-do: Decide whether range storage is required for group
+  // storage
+  size_t linear_tid = item.get_local_linear_id();
+  OutputIteratorT workitem_itr = block_itr + (linear_tid * ITEMS_PER_WORK_ITEM);
+#pragma unroll
+  for (uint32_t idx = 0; idx < ITEMS_PER_WORK_ITEM; idx++) {
+    workitem_itr[idx] = items[idx];
+  }
+}
+
+/// Stores a striped arrangement of work items linear segment of items.
+template <size_t ITEMS_PER_WORK_ITEM, typename InputT, typename OutputIteratorT,
+          typename Item>
+__dpct_inline__ void store_striped(const Item &item, OutputIteratorT block_itr,
+                                   InputT (&items)[ITEMS_PER_WORK_ITEM]) {
+
+  // This implementation does not take in account range storage across
+  // workgroup items To-do: Decide whether range storage is required for group
+  // storage
+  size_t linear_tid = item.get_local_linear_id();
+  OutputIteratorT workitem_itr = block_itr + linear_tid;
+  size_t GROUP_WORK_ITEMS = item.get_local_range().size();
+#pragma unroll
+  for (uint32_t idx = 0; idx < ITEMS_PER_WORK_ITEM; idx++) {
+    workitem_itr[(idx * GROUP_WORK_ITEMS)] = items[idx];
+  }
+}
+
+/// Stores a warp-striped arrangement of work items linear segment of items.
```

Comment:
```suggestion
/// Stores a subgroup-striped arrangement of work items linear segment of items.
```
Lets keep our terminology consistent.